store/RedisStore.js

```diff
@@ -20,13 +26,20 @@ const lua = `
   if ttl == -1 or (continueExceeding and current > max) then
     redis.call('PEXPIRE', key, timeWindow)
     ttl = timeWindow
+  
+  -- If the key is new or if its incremented value has exceeded the max value and exponential backoff is enabled then set its TTL
+  elseif ttl == -1 or (exponentialBackoff and current > max) then
```

Comment:
I'll merge this but you can remove the `ttl == -1` here since the first condition will handle that condition

```suggestion
  elseif exponentialBackoff and current > max then
```